src/front/wgsl/mod.rs

```diff
@@ -1004,13 +1004,16 @@ impl std::error::Error for ParseError {
 pub struct Parser {
     scopes: Vec<Scope>,
     lookup_type: FastHashMap<String, Handle<crate::Type>>,
+    /// Map of expressions that have associated variable names
+    named_expressions: FastHashMap<Handle<crate::Expression>, String>,
```

Comment:
Can we avoid this? It's a per-function thing, and keeping it at "global" state here is a bit undesired. Instead, when parsing a function, a hashmap needs to be created on the stack, then passed to the `StatementContext`. After the function is done parsing, it would just be moved into the produced function.